R/topics.R

```diff
@@ -51,21 +52,22 @@ topics_create <- function(name,
                           satisfies_pzs = NULL,
                           message_storage_policy = NULL,
                           schema_settings = NULL,
-                          message_retention_duration = NULL) {
+                          message_retention_duration = NULL,
+                          project = Sys.getenv("GCP_PROJECT")) {
   
   if(!is.null(message_retention_duration)) {
     message_retention_duration <- paste0(message_retention_duration, "s")
   }
   
-  topic_name <- as.topic_name(name)
+  topic_name <- as.topic_name(name, project = project)
```

Comment:
No need to use this here as project will be taken in `as.topic_name` by calling `ps_project_get()`.
Sorry this wasn't clear as it was only deployed in the `dev` versionl. Let me know if this accomodates your workflow.

Reply:
Ah, that's fair.  I figured someone may want to do fast switching with specifying project directly.  I had added that to all the functions in this PR.  If too much to maintain or not good API dev, you can disregard.

Reply:
I've willingly decided to keep project settings out of function arguments. I believe this leads to cleaner functions while still addressing the majority of possible workfflows. This is aligned to Mark Edmonson's [packages][1] (which have been of great inspiration for `googlePubsubR`).

[1]: https://code.markedmondson.me/googleCloudRunner/reference/cr_project_set.html